astropy/io/fits/connect.py

```diff
@@ -162,7 +163,10 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    # Read time coordinates as Time
+    hdr = FITS_time().read_time(table.header, t)
```

Comment:
The class was so that everything related to time could be kept in one place (@mhvk had suggested this would be better). Also, if you look at all the functions used within read_time, they are used to maintain a time state pertaining to an object.